internal/migration/migration.go

```diff
@@ -113,6 +113,11 @@ func (e *ModelExporter) Export(ctx context.Context, model description.Model) (de
 	if err := coordinator.Perform(ctx, e.scope, model); err != nil {
 		return nil, errors.Trace(err)
 	}
+	// The model now contains all the exported data from the legacy state along
+	// with the new domains' one. Time to validate.
+	if err := model.Validate(); err != nil {
+		return nil, errors.Trace(err)
+	}
```

Comment:
We should add a test for this, to ensure it's triggered.

Reply:
In the last commit I did a bit of refactoring so we can inject dependencies on the internal migration package, and therefore be able to correctly test this method (at least the Validation stage).

So the test is now there.